helpers/parallel.py

```diff
@@ -0,0 +1,100 @@
+import dataclasses
+from typing import Self
+
+from rollouts import (
+    FULLY_PARALLEL_UPLOAD_PROCESSING_BY_REPO,
+    PARALLEL_UPLOAD_PROCESSING_BY_REPO,
+)
+
+
+@dataclasses.dataclass(frozen=True)
+class ParallelProcessing:
+    """
+    This encapsulates Parallel Upload Processing logic
+
+    Upload Processing can run in essentially 4 modes:
+    - Completely serial processing
+    - Serial processing, but running "experiment" code (`is_experiment_serial`):
```

Comment:
I'm not super clear about what the difference is between `is_experiment_serial` and `is_experiment_parallel`.

Reply:
When doing the "experiment", the processor and finisher task run twice for the same upload.
There is also an example flow showing those two runs, and which flags they use.

How can I make this clearer?

Reply:
Hmm, I think my main confusion is around their dependency (i.e. you can't have both of them be true at the same time) and about how serial is "experimental" (because it's the default option right now). Looking at the code, I guess the "default" serial path is separate from the experiment, so there is (in some paths) a distinction on "experiment" serial and "default" serial. 

To address the issue of both variables being dependent, I'm wondering if it'll be clearer if we used an enum called "ExperimentMode" or something. This way, they cannot be true at the same time, and we can just use the same variable to distinguish these cases.